### ISSUE TITLE
Allow executions to have some kind of name/label

### ISSUE THREAD

**Issue by alkemist** [ratpack/ratpack#1027]
Something like Execution.current().name("doing-stuff")

**Comment by tomdcc** [comment]
Would definitely be helpful for the name to be available in an initializer, so that other code that wants to populate its registry in an initializer has access to the name. So should be on `ExecSpec` as well to allow `Execution.fork().name("foo")`

**Comment by ldaley** [comment]
The only thing really stopping us doing this is deciding whether a single opaque string is good enough. Some alternatives might be:

1. Arbitrary object (expected to have a reasonable toString())
2. `Map<String,String>`

I am leaning towards a simple `String` `name` property though.

**Comment by tomdcc** [comment]
Me too. Pretty easy to imagine an arbitrary (and mutable) descriptor object being abused instead of just putting stuff in the registry. Easy enough to call `.name(constructDescriptor('foo', 123, 'bar))` if you need something more structured.

**Comment by ldaley** [comment]
Yeah, let's just settle on a single `CharSequence` property. That leaves the door a little open for someone to do something fancy. I think we should call this property `identifier` or `label`. 

It should go on `ExecSpec` and default to _N_, where that is the _N_th execution forked from that execution. Each `Execution` can maintain a counter for this and `ExecControllerInternal` can provide a counter for top level executions.

Anyone want volunteer to implement?

**Comment by tomdcc** [comment]
I'm happy to but can't promise anything over the next couple of weeks.
